extensions/wikia/Email/EmailController.class.php

```diff
@@ -396,6 +412,106 @@ protected function getSocialMessages() {
 		];
 	}
 
+	private function generateMobileApplicationsBadges() {
+		$mobileApplicationsLinks = $this->generateMobileApplicationsLinks();
+		$hasMobileApplicationBadges = count( $mobileApplicationsLinks ) > 0;
+		$badges = null;
+
+		if ( $hasMobileApplicationBadges ) {
+			$badges = [];
+
+			if ( $mobileApplicationsLinks[self::ANDROID_PLATFORM] ) {
+				$badges[self::ANDROID_PLATFORM] = [
+					'link' => $mobileApplicationsLinks[self::ANDROID_PLATFORM],
+					'src' => EmailMobileBadges::getBadgeFor( $this->targetLang, self::ANDROID_PLATFORM )
+				];
+			}
+			if ( $mobileApplicationsLinks[self::IOS_PLATFORM] ) {
+				$badges[self::IOS_PLATFORM] = [
+					'link' => $mobileApplicationsLinks[self::IOS_PLATFORM],
+					'src' => EmailMobileBadges::getBadgeFor( $this->targetLang, self::IOS_PLATFORM )
+				];
+			}
+		}
+
+		return $badges;
+	}
+
+	private function generateMobileApplicationsLinks() {
+		global $wgMemc;
+
+		$mobileApplicationsLinks = $wgMemc->get( $this->createCacheKeyForMobileApplicationsLinks() );
+
+		if ( is_bool( $mobileApplicationsLinks )) {
+			$mobileApplicationsLinks = [];
+			$response = $this->fetchMobileApplicationsDetails();
+
+			if ( $response && $this->applicationsExistFor( $this->wg->CityId, $response )) {
```

Comment:
Also, it doesn't look like we need to support calling `applicationsExistFor()` for wiki IDs other than the current wiki ID, but if for some reason we do, we could make it the second arg and default to `$this->wg->CityId` in the function.